pkg/controller/azure/provider/resourcegroup.go

```diff
@@ -118,31 +178,34 @@ func (r *ResourceGroupReconciler) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, err
 	}
 
-	// look up the provider information for this instance
-	provider := &v1alpha1.Provider{}
-	providerNamespacedName := apitypes.NamespacedName{
-		Namespace: instance.Namespace,
-		Name:      instance.Spec.ProviderRef.Name,
-	}
-	if err := r.Get(ctx, providerNamespacedName, provider); err != nil {
-		return r.fail(instance, errorFetchingAzureProvider, fmt.Sprintf("failed to get provider on behalf of resource group %+v: %+v", providerNamespacedName, err))
-	}
-
 	// Retrieve azure client from the given provider config
-	azureClient, err := azureclient.NewClient(provider, r.kubeclient)
+	azureClient, err := r.connect(instance)
 	if err != nil {
 		return r.fail(instance, errorCreatingClientRG, err.Error())
 	}
 
-	// Create or Update ResourceGroup
-	// TODO: Do we want to run the Create / Update on every reconciliation?
-	if err := r.create(azureClient, instance.Spec.Name, instance.Spec.Location); err != nil {
-		return r.fail(instance, errorCreatingClientRG, err.Error())
+	// Check for deletion
+	if instance.DeletionTimestamp != nil {
+		if instance.Status.Condition(corev1alpha1.Deleting) == nil {
```

Comment:
My understanding is that this will check whether the instance has ever been in state deleting. If we're okay instead checking whether the instance is _currently_ in state Deleting perhaps we should add an `IsDeleting()` helper to `ConditionedStatus` similar to the existing `IsReady()` and `IsFailed()` helpers?

Reply:
https://github.com/crossplaneio/crossplane/blob/d95025d/pkg/controller/aws/rds/rdsinstance.go#L286

I also notice that newer controllers don't seem to test for the case when an instance is already deleting at all, so maybe we can just leave it out? Delete calls don't tend to trigger a requeue so as long as the delete finishes inside a minute (our current reconcile loop frequency) it should be okay? I could see a case either way here so I don't feel strongly.